### PR TITLE
Giza olympia fix bundled video creation and nft issuance

### LINKED ISSUE #3107: Bundled video creation and nft issuance
# Background
This issue _must_ be done after this one https://github.com/Joystream/joystream/issues/3106, which has the same sprit.

kdembler identified the video creation extrinsic currently does not allow automatic issuing of an NFT with a given NFT state, as seen from this signature

```Rust
pub fn create_video(
    origin,
    actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
    channel_id: T::ChannelId,
    params: VideoCreationParameters<T>,
)
```

# Proposal
Allow bundled creation+issuance+initial NFT auction state, by updating `VideoCreationParametersRecord` to

```Rust

pub struct VideoCreationParametersRecord<NewAssets> {
    ...
    /// Automated NFT issuance
    pub auto_issue_nft: Option<AutomatedIssueNFTParameters>
}

pub struct AutomatedIssueNFTParameters {
    non_channel_owner: Option<T::MemberId>, 
    royalty: Option<Royalty>,
    nft_metadata: Metadata,
    auto_setup_transactional_status: AutomatedTransactionalStatusSetupParameters // defined in https://github.com/Joystream/joystream/issues/3106
}
```

With obvious semantics implications for `create_video`, and remember to update

* event signature of event `VideoCreated`  with exactly same considerations as for `NFTIssued` in https://github.com/Joystream/joystream/issues/3106.
* handbook

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201709380879215/1201717130441449) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by dobertRowneySr**
fixes:
- #3107
- #3106 

I also removed all the unused code like `PersonId` and the likes that was not removed this far, in order to have cleaner type aliases

~*Update 2022-02-02*: rebased to current `olympia` . This lead to a larger number of commits
Relevant comits start from : c86b9ddb8029437e854b166797e513b5efd6d703~
Relevant code section are in the `content` module, in particular:
- `NFTIssuanceParameter` type introduced
- `InitTransactionalStatus` type introduced
- `ensure_valid_init_transactional_status` guard added
- `create_video` & `issue_nft` extrinsic argument modified

### CODE REVIEW

File: runtime/src/constants.rs [hunk 1]
```diff
@@ -245,7 +245,6 @@ lazy_static! {
         ].to_vec()),
         // Bounty
         (BountyLockId::get(), [
-            InvitedMemberLockId::get(),
```
Comment (shamil-gadelshin) [inline runtime/src/constants.rs]: Why did you remove that?

File: scripts/generate-weights.sh [hunk 2]
```diff
@@ -68,4 +68,4 @@ benchmark membership
 benchmark bounty
 benchmark blog
 benchmark joystream_utility
-benchmark storage_v2
```
Comment (shamil-gadelshin) [inline scripts/generate-weights.sh:71]: Comment the line, please. It produces an error.

File: runtime-modules/content/src/lib.rs [hunk 3]
```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    pub fn ensure_valid_init_transactional_status(
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs]: Do we need this method as public?

File: runtime-modules/content/src/lib.rs [hunk 4]
```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
...
+                let mut auction = AuctionRecord::new(params.clone());
+                if params.starts_at.is_none() {
+                    auction.starts_at = <frame_system::Module<T>>::block_number();
+                }
+                Ok(TransactionalStatus::Auction(auction))
+            }
+        }
+    }
+
+    pub fn construct_owned_nft(
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs:2113]: Do we need this method as public?

File: runtime-modules/content/src/lib.rs [hunk 5]
```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    pub fn ensure_valid_init_transactional_status(
+        init_status: &InitTransactionalStatus<T>,
+    ) -> Result<
+        TransactionalStatus<
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs]: It makes sense to introduce an alias for such a complex type.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
The underlying issues look addressed. However, there are multiple issues with the code:
 - no doc-comments for public types, extrinsics, and methods
 - please, remove `allow all` clippy warning statement and fix all clippy issues in the pallet
 - found this line in the code: `// TODO: Solve #now` 
 - rebase seems unsuccessful: 1) there is still unrelated code in the PR (removed empty extrinsics), 2) `generate_weights.sh` script fails now (incorrect benchmark name)
 - there are public methods with no external usage (should be private)

File: runtime/src/constants.rs (hunk 1 above)
Comment (dobertRowneySr) [reply runtime/src/constants.rs]: I am not sure why github as that. It is marked as outdated, and the line is present on my code

File: scripts/generate-weights.sh (hunk 2 above)
Comment (dobertRowneySr) [reply scripts/generate-weights.sh:71]: fixed in 8184b6caecc3bf303b9afd0d9f1ee52464ef0597

File: runtime-modules/content/src/lib.rs (hunk 3 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs]: fixed in 92a405cf94a779685a5ed4f648210345e00a75fa

File: runtime-modules/content/src/lib.rs (hunk 4 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs:2113]: fixed in 92a405cf94a779685a5ed4f648210345e00a75fa

File: runtime-modules/content/src/lib.rs (hunk 5 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs]: fixed in afcde3072cf9c430da415bf088c3e96196d7f02b

**Comment by dobertRowneySr** [comment]
> The underlying issues look addressed. However, there are multiple issues with the code:
> 
>     * no doc-comments for public types, extrinsics, and methods
> 
>     * please, remove `allow all` clippy warning statement and fix all clippy issues in the pallet
> 
>     * found this line in the code: `// TODO: Solve #now`
> 
>     * rebase seems unsuccessful: 1) there is still unrelated code in the PR (removed empty extrinsics), 2) `generate_weights.sh` script fails now (incorrect benchmark name)
> 
>     * there are public methods with no external usage (should be private)

- the `TODO` was an old reference to myself that I solved and forgot to remove
- benchmark should be fixed
- removed the cargo directive 
- added doc-comments on types and helper functions added

I could locate the `removed_empty_extrinsic` reference in the Github commit tab

File: runtime-modules/content/src/lib.rs [hunk 6]
```diff
@@ -1031,54 +962,6 @@ decl_module! {
 
             Self::deposit_event(RawEvent::VideoCategoryDeleted(actor, category_id));
         }
-
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs:1034]: Isn't it a rebase artifact?

Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs:1034]: I am not following. The BASE commit in olympia has this code. In this PR that I am opening I removed it

Comment (shamil-gadelshin) [reply runtime-modules/content/src/lib.rs:1034]: Ok. So, it's part of the PR. I was expecting only the NFT-related code.

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Pull request merged**